homeassistant/components/light/tplink.py

```diff
@@ -150,15 +150,15 @@ def update(self):
                 self._rgb = hsv_to_rgb(self.smartbulb.hsv)
             if self.smartbulb.has_emeter:
                 self._emeter_params[ATTR_CURRENT_CONSUMPTION] \
-                    = "%.1f W" % self.smartbulb.current_consumption()
+                    = "%.1f W" % (self.smartbulb.current_consumption() / 1e3)
```

Comment:
Storing the unit together with the value is bad design from the beginning. The unit should rather be part of the attribute name. See eg the switch base entity. Since this is a breaking change anyway, I suggest we make it right now.